### PR TITLE
Use DT labels for UART driver names (based on PR #208)

### PR DESCRIPTION
**Pull request by galak**
Convert over drivers that are used on DTS platforms to use a device tree label as the means to generate the driver instance name.  This is based on PR #208.

### CODE REVIEW

**Review by fallrisk — APPROVED** [review]

File: dts/common/yaml/uart.yaml [hunk 1]
```diff
@@ -22,4 +22,9 @@ properties:
       category: required
       description: Clock gate information
       generation: define
+  - label:
+      type: string
+      category: optional
+      description: Human readable string describing the device
```
Comment (pfalcon) [inline dts/common/yaml/uart.yaml]: Is this description correct? I'd say in usage here it's "API identifier of the device".

**Review by pfalcon — APPROVED** [review]
Perhaps a minor clarification in the bindings doc.

**Review by carlescufi — APPROVED** [review]

**Comment by erwango** [comment]
Adding @atorgue for stm32 dts file review

**Review by erwango — APPROVED** [review]

File: dts/common/yaml/uart.yaml (hunk 1 above)
Comment (galak) [inline dts/common/yaml/uart.yaml]: Best comment on this on PR #208.  However I'm leaving the description vague as its the Zephyr usage of the label that is for API identifier of the device.

**Comment by erwango** [comment]
One question: Will it be ok for vendors to modify names for a specific reason (for instance specific use in Linux)? Or is that PR binding to a zephyr naming convention ?
Reason for this question is to understand possible constraints raised on Linux side (field is not used today, but it might be in future, and we'd like to understand the nature of the constraint)

File: drivers/serial/Kconfig.stm32 [hunk 2]
```diff
@@ -99,14 +67,6 @@ config UART_STM32_PORT_5
 	  Enable support for UART5 port in the driver. Say y here
 	  if you want to use UART5 device.
 
-config UART_STM32_PORT_5_NAME
```
Comment (Fabo) [inline drivers/serial/Kconfig.stm32]: You forgot to remove UART_STM32_PORT_6_NAME below

**Review by Fabo — CHANGES_REQUESTED** [review]

File: drivers/serial/Kconfig.stm32 (hunk 2 above)
Comment (galak) [inline drivers/serial/Kconfig.stm32]: Good catch, will fix.

**Comment by galak** [comment]
I expect someone can change the label as they see fit, no different than someone setting the Kconfig name to whatever they wanted.

**Review by Fabo — APPROVED** [review]

**Comment by MaureenHelm** [comment]
Rebased after merging #208

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**